jenkins/linux_c_option_test.sh

```diff
@@ -56,6 +56,7 @@ declare -a arr=(
     "-Drun_longhaul_tests=ON"
     "-Duse_prov_client=ON -Dhsm_custom_lib=$custom_hsm_lib"
     "-Drun_e2e_tests=ON -Drun_sfc_tests=ON -Duse_edge_modules=ON"
+    "-Drun_e2e_tests=ON -DUSE_BALTIMORE_CERT=ON"
```

Comment:
Do we need this?

Reply:
This was added to address Dane's concern that new #defines should be tested.

Reply:
When running cmake from the command line, it is possible to specify command line options to cmake that will set values in the cache. This is done with a -DVARIABLE:TYPE=VALUE syntax on the command line. This is useful for non-interactive nightly test builds.
​
https://cmake.org/runningcmake/
​
-D defines variables for cmake, not #defines in the project

that is done with add_definition()

so passing that actually gets ignored by cmake, since it's not used anywhere